extension/chrome/src/js/background.js

```diff
@@ -135,82 +129,65 @@ function retrieveActivities() {
-		  return data;
+		const buildUrlPR = function(url, role){
+			return url + role;
 		}
-		var authorDefered = jQuery.Deferred()
-		var resolveAuthor = function(data) {
-		  authorDefered.resolve();
-		  return data;
-		}
-		var mergeResults = function(data){
-			jQuery.merge(allPR, data.values)
-		};
-		var rerunRequest = function(role) {
+		const rerunRequest = function(role) {
 			return function(err) {
-				var resolveDeferred = role === 'reviewer' ? resolveReviewers : resolveAuthor;
-				if(err.status == 404) {
-					return jQuery
-					.get(buildUrlPR(urlPR, role))
-					.then(mergeResults)
-					.then(resolveDeferred);
+				if (err.status == 404) {
+					return fetch(buildUrlPR(urlPR, role))
 				}
+				return Promise.reject(err)
 			}
 		};
-		var rerunRequestReviewers = rerunRequest('reviewer');
-		var rerunRequestAuthor = rerunRequest('author');
-
-		jQuery
-		.get(buildUrlPR(urlPRNew, 'reviewer'))
-		.then(mergeResults)
-		.then(resolveReviewers)
-		.fail(rerunRequestReviewers);
-
-		jQuery
-		.get(buildUrlPR(urlPRNew, 'author'))
-		.then(mergeResults)
-		.then(resolveAuthor)
-		.fail(rerunRequestAuthor);
-
-		jQuery.when(reviewersDefered, authorDefered).done(function(){
-			var activities = [];
-			var requests = [];
+		const rerunRequestReviewers = rerunRequest('reviewer');
+		const rerunRequestAuthor = rerunRequest('author');
+
+		const reviewersPromised = fetch(buildUrlPR(urlPRNew, 'reviewer'))
+			.then(res => res.json())
+			.catch(rerunRequestReviewers);
+		
+		const authorPromised = fetch(buildUrlPR(urlPRNew, 'author'))
+			.then(res => res.json())
+			.catch(rerunRequestAuthor);
+
+		Promise.all([reviewersPromised, authorPromised]).then(function(results) {
+			if (results[0].errors) { throw results[0].errors }
+			if (results[1].errors) { throw results[1].errors }
+			const allPR = results[0].concat(results[1])
+			let activities;
+			const requests = [];
 			// loop through PRs and request activities
 			allPR.forEach(function(pr){
-				var prLink = '';
+				let prLink = '';
 				if(pr.links && pr.links.self) {
 					prLink = pr.links.self[0].href.replace(items.currentStashBaseUrl, '');
-					prLink = items.currentStashBaseUrl + '/rest/api/1.0' + prLink + '/activities?avatarSize=64';
+					prLink = `${items.currentStashBaseUrl  }/rest/api/1.0${  prLink  }/activities?avatarSize=64`;
 				}
 
 				if(prLink) {
-					requests.push(jQuery.get(prLink)
-					.done(function(activityList){
+					requests.push(fetch(prLink)
+						.then(res => res.json())
+						.then(function(activityList){
 						// get comments after PR was updated
-						jQuery.each(activityList.values, function(index, activity){
-							jQuery.extend(activity, { pullrequest: pr });
-							activities.push(activity);
-						});
-					}));
+							activities = activityList.values.map(function(activity){
+								return Object.assign(activity, { pullrequest: pr });
+							});
+						})
+						.catch(err => { console.error(`Request to ${  prLink  } failed`, err)})
```

Comment:
[minor] might be worth documenting that this `catch` returns `Promise.resolve(undefined)` so that `Promise.all(requests)` is guaranteed to be never rejected (= equivalent to `jQuery.deferred.always`)